packages/query/src/query/useList.ts

```diff
@@ -366,29 +367,52 @@ export const getFetchListQueryFn = (props: UseListProps) => {
       lte,
       not,
       select,
-      contains,
+      contains
     } = listFilters
 
+    const getSelectString = () => {
+      if (countOnly) {
+        return module?.select?.count || '*'
+      }
+
+      // @ts-ignore
+      const filterSelect = module?.select?.filter
+      if (filters?.length && filterSelect) {
+        return filterSelect
+      }
+
+      return select || module?.select?.list || '*'
+    }
+
```

Comment:
This PR enhances CrudModule.select with CrudModule.select.filter?: string . Using this new key will allow downstream apps to dedicate a select statement specifically for filtering in the admin panel. This select statement will be picked up by useList nested within CrudHeader to automatically set a different query for filtering instead of the default select. This is so that we can apply !inner to the filter select over the default select.